File: mycollab-web-premium/src/main/java/com/esofthead/mycollab/premium/module/project/view/problem/ProblemTableFieldDef.java
package com.esofthead.mycollab.premium.module.project.view.problem;

import com.esofthead.mycollab.common.TableViewField;
import com.esofthead.mycollab.common.i18n.GenericI18Enum;
import com.esofthead.mycollab.module.project.i18n.ProblemI18nEnum;
import com.esofthead.mycollab.vaadin.ui.UIConstants;

/**
 * 
 * @author MyCollab Ltd.
 * @since 1.0
 */
public interface ProblemTableFieldDef {

	public static TableViewField selected = new TableViewField(null,
			"selected", UIConstants.TABLE_CONTROL_WIDTH);

	public static TableViewField action = new TableViewField(null, "id");

	public static TableViewField name = new TableViewField(
			ProblemI18nEnum.FORM_NAME, "issuename",
			UIConstants.TABLE_X_LABEL_WIDTH);

	public static TableViewField description = new TableViewField(
			GenericI18Enum.FORM_DESCRIPTION, "description",
			UIConstants.TABLE_EX_LABEL_WIDTH);

	public static TableViewField raisedby = new TableViewField(
			ProblemI18nEnum.FORM_RAISED_BY, "raisedByUserFullName",
			UIConstants.TABLE_X_LABEL_WIDTH);

	public static TableViewField assignUser = new TableViewField(
			GenericI18Enum.FORM_ASSIGNEE, "assignedUserFullName",
			UIConstants.TABLE_X_LABEL_WIDTH);

	public static TableViewField datedue = new TableViewField(
			ProblemI18nEnum.FORM_DATE_DUE, "datedue",
			UIConstants.TABLE_DATE_WIDTH);

	public static TableViewField status = new TableViewField(
			ProblemI18nEnum.FORM_STATUS, "status",
			UIConstants.TABLE_S_LABEL_WIDTH);

	public static TableViewField impact = new TableViewField(
			ProblemI18nEnum.FORM_IMPACT, "impact",
			UIConstants.TABLE_S_LABEL_WIDTH);

	public static TableViewField priority = new TableViewField(
			ProblemI18nEnum.FORM_PRIORITY, "priority",
			UIConstants.TABLE_S_LABEL_WIDTH);

	public static TableViewField rating = new TableViewField(
			ProblemI18nEnum.FORM_RATING, "level",
			UIConstants.TABLE_M_LABEL_WIDTH);
}

File: mycollab-web-pro/src/main/scala/com/esofthead/mycollab/premium/module/project/view/problem/ProblemTableFieldDef.scala
package com.esofthead.mycollab.premium.module.project.view.problem

import com.esofthead.mycollab.common.TableViewField
import com.esofthead.mycollab.common.i18n.GenericI18Enum
import com.esofthead.mycollab.module.project.i18n.ProblemI18nEnum
import com.esofthead.mycollab.vaadin.ui.UIConstants

/**
 * @author MyCollab Ltd
 * @since 5.0.9
 */
object ProblemTableFieldDef {
  val selected: TableViewField = new TableViewField(null, "selected", UIConstants.TABLE_CONTROL_WIDTH)
  val action: TableViewField = new TableViewField(null, "id", -1)
  val name: TableViewField = new TableViewField(ProblemI18nEnum.FORM_NAME, "issuename", UIConstants.TABLE_X_LABEL_WIDTH)
  val description: TableViewField = new TableViewField(GenericI18Enum.FORM_DESCRIPTION, "description", UIConstants.TABLE_EX_LABEL_WIDTH)
  val raisedby: TableViewField = new TableViewField(ProblemI18nEnum.FORM_RAISED_BY, "raisedByUserFullName", UIConstants.TABLE_X_LABEL_WIDTH)
  val assignUser: TableViewField = new TableViewField(GenericI18Enum.FORM_ASSIGNEE, "assignedUserFullName", UIConstants.TABLE_X_LABEL_WIDTH)
  val datedue: TableViewField = new TableViewField(ProblemI18nEnum.FORM_DATE_DUE, "datedue", UIConstants.TABLE_DATE_WIDTH)
  val status: TableViewField = new TableViewField(ProblemI18nEnum.FORM_STATUS, "status", UIConstants.TABLE_S_LABEL_WIDTH)
  val impact: TableViewField = new TableViewField(ProblemI18nEnum.FORM_IMPACT, "impact", UIConstants.TABLE_S_LABEL_WIDTH)
  val priority: TableViewField = new TableViewField(ProblemI18nEnum.FORM_PRIORITY, "priority", UIConstants.TABLE_S_LABEL_WIDTH)
  val rating: TableViewField = new TableViewField(ProblemI18nEnum.FORM_RATING, "level", UIConstants.TABLE_M_LABEL_WIDTH)
}
